modules/git/command.go

```diff
@@ -140,7 +149,12 @@ func (c *Command) RunWithContext(rc *RunContext) error {
 
 	desc := c.desc
 	if desc == "" {
-		desc = fmt.Sprintf("%s %s [repo_path: %s]", c.name, strings.Join(c.args, " "), rc.Dir)
+		args := make([]string, len(c.args))
+		copy(args, c.args)
+		for _, urlArgIndex := range c.urlArgIndexes {
+			args[urlArgIndex] = util.NewStringURLSanitizer(args[urlArgIndex], true).Replace(args[urlArgIndex])
+		}
+		desc = fmt.Sprintf("%s %s [repo_path: %s]", c.name, strings.Join(args, " "), rc.Dir)
```

Comment:
Although it looks not bad, why do not we use a more general method to grep and replace the URL-like strings in `c.args` for the final `desc`?

The `AddURLArgument` makes code more complex, and I can bet that one day in future, some other code containing sensitive URL would come into code base without calling this inconspicuous `AddURLArgument`.

Reply:
I think the current method has a better performance than detecting every arguments when most time, there is no any URL in the args.

Reply:
To detect URLs, a `strings.Index("://")` for each arg won't cost much, it won't cause any performance problem.

And if the `AddURLArgument` is used, how can it be guaranteed that:
1. All git commands with URLs have been processed without omission now?
2. All the PRs in future can call AddURLArgument correctly without mistake?

Reply:
For git command, only serval commands like `clone`, `remote add`, `pull`, `push` will use URLs. So we just need to change these command functions. I don't think there are many of them.

Reply:
That's the `current situation`, how about question 2, the future? patch them again and again?

If every PR only considers current situation, the code base will become more and more dirty. That's how bug comes, and that's why some modules become more and more difficult to be maintained.

Reply:
If you do not mind, I can provide a more general (and faster) URL sanitizer, and improve the `sanitizedError` to support `Unwrap`, etc.

Reply:
What's your idea? How to make it faster?

Reply:
Commented below